infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationSourceManager.java

```diff
@@ -247,17 +226,75 @@ public LogReplicationEntry receive(LogReplicationEntry message) {
         // Process ACKs from Application, for both, log entry and snapshot sync.
         if(message.getMetadata().getMessageMetadataType() == MessageType.LOG_ENTRY_REPLICATED) {
             log.debug("Log entry sync ACK received on timestamp {}", message.getMetadata().getTimestamp());
+            lastAckedTimestamp.set(message.getMetadata().getTimestamp());
```

Comment:
Actually, this is a bit weird. But I realized it the other day while looking at other things. This method for the Source is not really ever called. All the logic for processing ACKs is taken care in the SnapshotSender and LogEntrySender respectively, so it is there where we should be setting this lastAckedTimestamp.

Reply:
Actually, I think we have a major concern here and its that SnapshotSender does not really accumulate intermediate SNAPSHOT_REPLICATED, so it only updates at the end.. But you can bypass that (for now) by updating when you send something...